File: BehsaApp/BehsaApp.Web/Modules/ServerTypes/Behsa/ProvincesRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface ProvincesRow {
    Id?: number;
    ProvinceName?: string;
}

export abstract class ProvincesRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'ProvinceName';
    static readonly localTextPrefix = 'Behsa.Provinces';
    static readonly lookupKey = 'Behsa.Provinces';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<ProvincesRow>('Behsa.Provinces') }
    static async getLookupAsync() { return getLookupAsync<ProvincesRow>('Behsa.Provinces') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<ProvincesRow>();
}

File: BehsaApp/BehsaApp.Web/Modules/Behsa/Provinces/ProvincesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace BehsaApp.Behsa;

[ConnectionKey("Behsa"), Module("Behsa"), TableName("Provinces")]
[DisplayName("Provinces"), InstanceName("Provinces")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed class ProvincesRow : Row<ProvincesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Province Name"), Size(128), NotNull, QuickSearch, NameProperty]
    public string ProvinceName
    {
        get => fields.ProvinceName[this];
        set => fields.ProvinceName[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField ProvinceName;

    }
}